drivers/at86rf2xx/at86rf2xx_rf_ops.c

```diff
@@ -0,0 +1,971 @@
+{
+    at86rf2xx_t *dev = hal->priv;
+    mutex_unlock(&dev->lock);
+    hal->cb(hal, ev);
+    mutex_lock(&dev->lock);
+}
+
+static inline void _isr_recv_complete(ieee802154_dev_t *hal)
+{
+    at86rf2xx_t *dev = hal->priv;
+
+    if (IS_ACTIVE(AT86RF2XX_BASIC_MODE)) {
+        uint8_t phy_status = at86rf2xx_reg_read(dev, AT86RF2XX_REG__PHY_RSSI);
+        bool crc_ok = phy_status & AT86RF2XX_PHY_RSSI_MASK__RX_CRC_VALID;
+
+        if (crc_ok) {
+            _dispatch_event(hal, IEEE802154_RADIO_INDICATION_RX_DONE);
+        }
+        else {
+            _dispatch_event(hal, IEEE802154_RADIO_INDICATION_CRC_ERROR);
+        }
+    }
+    else {
+        _dispatch_event(hal, IEEE802154_RADIO_INDICATION_RX_DONE);
+    }
+}
+
+void at86rf2xx_irq_handler(ieee802154_dev_t *hal)
+{
+    at86rf2xx_t *dev = hal->priv;
+    uint8_t irq_mask;
+    uint8_t state;
+
+    /* If transceiver is sleeping register access is impossible and frames are
+     * lost anyway, so return immediately.
+     */
+    mutex_lock(&dev->lock);
+    state = at86rf2xx_get_status(dev);
+    if (state == AT86RF2XX_STATE_SLEEP) {
+        mutex_unlock(&dev->lock);
+        return;
+    }
+
+    /* read (consume) device status */
+    irq_mask = at86rf2xx_reg_read(dev, AT86RF2XX_REG__IRQ_STATUS);
+
+    if (irq_mask & AT86RF2XX_IRQ_STATUS_MASK__RX_START) {
+        _dispatch_event(hal, IEEE802154_RADIO_INDICATION_RX_START);
+        DEBUG("[at86rf2xx] EVT - RX_START\n");
+    }
+
+    if (irq_mask & AT86RF2XX_IRQ_STATUS_MASK__TRX_END) {
+        if ((state == AT86RF2XX_PHY_STATE_RX)
+            || (state == AT86RF2XX_PHY_STATE_RX_BUSY)) {
+            DEBUG("[at86rf2xx] EVT - RX_END\n");
+
+            _isr_recv_complete(hal);
+
+        }
+        else if (state == AT86RF2XX_PHY_STATE_TX) {
+            _dispatch_event(hal, IEEE802154_RADIO_CONFIRM_TX_DONE);
+        }
+    }
+
+    if (irq_mask & AT86RF2XX_IRQ_STATUS_MASK__CCA_ED_DONE) {
+        _dispatch_event(hal, IEEE802154_RADIO_CONFIRM_CCA);
+    }
+
+    mutex_unlock(&dev->lock);
+}
+
+static void _event_handler(event_t *event)
+{
+    at86rf2xx_bhp_ev_t *bhp = container_of(event, at86rf2xx_bhp_ev_t, ev);
+    ieee802154_dev_t *hal = bhp->hal;
+    at86rf2xx_irq_handler(hal);
+}
+
+void _irq_handler(void *ctx)
+{
+    at86rf2xx_bhp_ev_t *bhp = ctx;
+    event_post(bhp->evq, &bhp->ev);
+}
+
+int at86rf2xx_init_event(at86rf2xx_bhp_ev_t *bhp, const at86rf2xx_params_t *params,
+                         ieee802154_dev_t *hal, ieee802154_sec_context_t *sec, event_queue_t *evq)
+{
+    bhp->hal = hal;
+    bhp->evq = evq;
+    bhp->ev.handler = _event_handler;
+
+    return at86rf2xx_init(&bhp->dev, params, hal, sec, _irq_handler, bhp);
+}
+
+static const ieee802154_radio_ops_t at86rf2xx_ops = {
+    .caps =  IEEE802154_CAP_24_GHZ
+          | IEEE802154_CAP_IRQ_CRC_ERROR
+          | IEEE802154_CAP_IRQ_RX_START
+          | IEEE802154_CAP_IRQ_TX_DONE
+          //| IEEE802154_CAP_IRQ_CCA_DONE
```

Comment:
it is indeed valid, but I didn't implement it because I kind of copied the old version of this driver into the Radio HAL port. If OK, I can add it in a follow up. But I will definitely remove  the comment there.

Reply:
ignore my last comment, that was an old commit